src/main/java/io/neonbee/entity/EntityVerticle.java

```diff
@@ -250,34 +255,19 @@ private Future<Void> announceEntityVerticle(Vertx vertx) {
         return entityTypeNames()
                 .map(entityTypeNames -> entityTypeNames != null ? entityTypeNames : Set.<FullQualifiedName>of())
                 .compose(entityTypeNames -> {
-                    AsyncMap<String, Object> sharedMap = NeonBee.get(vertx).getAsyncMap();
                     List<Future<Void>> announceFutures =
                             entityTypeNames.stream().map(EntityVerticle::sharedEntityMapName).map(name -> {
-                                LOGGER.debug("Acquire lock {} for announcement of entity verticle", name);
-                                return new SharedDataAccessor(vertx, EntityVerticle.class).getLock(name)
-                                        .onFailure(throwable -> {
-                                            LOGGER.error("Error acquiring lock with name {}", name, throwable);
-                                        }).compose(lock -> {
-                                            return sharedMap.get(name).compose(qualifiedNamesOrNull -> {
-                                                String qualifiedName = getQualifiedName();
-                                                JsonArray qualifiedNames =
-                                                        qualifiedNamesOrNull != null ? (JsonArray) qualifiedNamesOrNull
-                                                                : new JsonArray();
-                                                if (!qualifiedNames.contains(qualifiedName)) {
-                                                    qualifiedNames.add(qualifiedName);
-                                                }
-
-                                                LOGGER.info(
-                                                        "Announce entity {} is served by entity verticle with qualified name {}",
-                                                        name, qualifiedName);
-                                                return sharedMap.put(name, qualifiedNames);
-                                            }).onComplete(anyResult -> {
-                                                LOGGER.debug("Releasing lock {}", name);
-                                                lock.release();
-                                            });
-                                        });
+                                String qualifiedName = getQualifiedName();
+                                return getRegistry(vertx).register(name, qualifiedName);
                             }).collect(Collectors.toList());
                     return allComposite(announceFutures).mapEmpty();
                 });
     }
+
+    static WriteSafeRegistry getRegistry(Vertx vertx) {
```

Comment:
You ensure thread safety via the locking in the `WriteSafeRegistry` itself, but honestly I don't feel comfortable with this singleton approach. It comes with several other drawbacks like the fact that we need to introduce a `EntityVerticleExtension`. I would even rather manage it centrally via the NeonBee instance instead. Did you consider this?

Reply:
Maybe, let's have a short sync after standup to discuss available options.